### ISSUE TITLE
Fix for webpack issue

### ISSUE THREAD

**Issue by Chathula** [node-facebook/facebook-node-sdk#82]
webpack.config.js -> install json-loader first (`npm install --save-dev json-loader`)

```javascript
// var path = require('path');

module.exports = {
//  entry: 'index',
//  output: {
//    path: path.join(__dirname, 'scripts'),
//    filename: 'bundle.js'
// },
  module: {
    loaders: [
      { test: /\.json$/, loader: 'json-loader' }
    ]
  },
  node: {
    console: true,
    fs: 'empty',
    net: 'empty',
    tls: 'empty'
  }
};
```

**Comment by dantman** [comment]
Issue? This is a server side sdk, where are you trying to use a webpack bundled version?

**Comment by Chathula** [comment]
this is a fix! for the people who want to use this with react, webpack!

**Comment by dantman** [comment]
Where? This is a server-side SDK, not a browser sdk. React Native has its own SDK and RN uses Browserify instead of Webpack anyways. React in the browser has Facebook's JS SDK.

**Comment by Chathula** [comment]
i used this with Next.js React.. i have to change webpack config like this to make it work.

**Comment by dantman** [comment]
Your request still makes no sense. Next.js is just a framework, you're still just running plain node.js on the server. This would make sense if Next.js decided to make an alternative webpack server bundle and execute that server side, but they explicitly say "Warning: Adding loaders to support new file types (css, less, svg, etc.) is not recommended because only the client code gets bundled via webpack and thus it won't work on the initial server rendering.".

In other words, you are loading facebook-node-sdk client side, which you explicitly should not be doing. Find a way to import `fb` so that is is ONLY imported in server-side code in Next.js.

**Comment by Chathula** [comment]
i have worked with next.js with fb module! :D that is why i wanted to share the fix for the people who want this with webpack! :D

**Comment by dantman** [comment]
And I'm saying that's not an answer to my question or the correct 'fix' for your problem. Next.js only uses Webpack for the client. `fb` is not supposed to be used on the client, hence it should **never** have been part of the webpack bundle in the first place. Meaning you required `fb` in a place you should not have been requiring it and instead of fixing the bug in your code, you hacked around it and instead just delivered a server side library to the browser. That is not a *fix*.

**Comment by hmontes** [comment]
Do you know an alternative for React? You can't import module with Facebook's JS SDK.

Thanks

**Comment by dantman** [comment]
@hmontes I have no clue what you're talking about, React is not a JS environment or bundler (node, browser, webpack, etc...) it's just a JS library.
